Fix scene tree drag & drop places node as child (#6912)

Keetz opened the pull request:
The meta "_editor_collapsed" is no longer a part of a node, so i have updated the code to use "is_displayed_folded()" instead.

If a node was dropped above a certain node, instead of just being placed above it, it would end up as a child to the node above the place you dropped it if that node is unfolded and has children. I have fixed this so it now just gets placed above the node even if the above node is unfolded.

See #6912 for bug reference.

Keetz committed:
Fix scene tree drag & drop places node as child (#6912)

akien-mga commented on tools/editor/scene_tree_dock.cpp:
```diff
@@ -1564,30 +1564,28 @@ static bool _is_node_visible(Node* p_node) {
 	for(int i=0;i<p_node->get_child_count();i++) {
-
 		Node* child = p_node->get_child(i);
+
```
Please avoid unnecessary line changes, they make the diff harder to read while they are not related to the functional change.

akien-mga commented on tools/editor/scene_tree_dock.cpp:
```diff
@@ -1564,30 +1564,28 @@ static bool _is_node_visible(Node* p_node) {
 static bool _has_visible_children(Node* p_node) {
 
-	bool collapsed = p_node->has_meta("_editor_collapsed") ? (bool)p_node->get_meta("_editor_collapsed") : false;
+    bool collapsed = p_node->is_displayed_folded();
```
You should use tabs for indentation, not spaces.

akien-mga commented:
Seems to work great from some quick testing.

akien-mga commented:
Could you explain a bit more why the code you removed is no longer necessary though?

Keetz commented:
@akien-mga The code i removed was causing a dropped node to be added as a child of the node above the one it was dropped on if it has any children. I created a small video example to show whats going on as it is kinda hard to explain.

![godot_dragndrop](https://cloud.githubusercontent.com/assets/6707330/20037916/755dce84-a42a-11e6-9dbe-cfdff3beed5a.gif)

I am dropping "DragMe" above "Child2" and therefore expect it to be placed right above it, but instead it is placed beneath "Child1" as a child. This should only have happened if i had dropped "Child2" directly ontop of "Child1".

So i refactored the code a little so a node dropped above another always gets the index of the node it was dropped above.

```
to_pos=to_node->get_index();
to_node=to_node->get_parent();
```

So with this fix it ends up like this:

![dragndrop_fix](https://cloud.githubusercontent.com/assets/6707330/20038058/f0d5256a-a42c-11e6-9e93-f7791f4e031e.png)

bojidar-bg commented:
@akien-mga Would be nice if we can merge this, since we are getting more and more duplicate reports :wink:
